src/pages/index.js

```diff
@@ -15,86 +16,215 @@ import {
+      .catch((err) => {
+        console.error(`Error on Card Like ${err}`);
+      });
+  }
+}
+
+//add card functionality
 const createCard = (data) => {
-  const card = new Card(data, "#card-template", handleImagePreview);
+  const card = new Card(
+    data,
+    "#card-template",
+    handleImagePreview,
+    handleDeleteCard,
+    handleLikeClick
+  );
   return card.getCardElement();
 };
 
-//render card elements
-const cardListEl = new Section(
-  {
-    items: initialCards,
-    renderer: (data) => {
-      cardListEl.addItem(createCard(data));
-    },
-  },
-
-  ".cards__list"
+const addCardPopup = new PopupWithForm(
+  "#add-card-modal",
+  handleAddCardFormSubmit
 );
+addCardPopup.setEventListeners();
+
+addNewCardButton.addEventListener("click", () => {
+  addCardPopup.open();
+});
 
-cardListEl.renderItems();
+function handleAddCardFormSubmit(data) {
+  const name = data.title.trim();
+  const link = data.link.trim();
+  addCardPopup.close();
+  addCardFormElement.reset();
+  addCardValidator.disableSubmitButton();
+  api
+    .createCard({ name, link })
+    .then((res) => {
+      const card = createCard(res);
+      section.addItem(card);
+    })
+    .finally(() => {
+      addCardPopup.setLoading(false);
+    });
+}
 
-//preview Image Modal
+//preview card image functionality
 const previewImageModal = new PopupWithImage("#preview-image-modal");
 previewImageModal.setEventListeners();
 
 function handleImagePreview(card) {
   previewImageModal.open(card);
 }
 
-//edit profile functionality
+//delete card functionality
+function handleDeleteCard(cardId, cardElement) {
+  deleteModalConfirmation.open(cardId, cardElement);
+  console.log(cardId, cardElement);
+}
+
+const deleteModalConfirmation = new PopupConfirmation(
+  "#modal-confirm",
+  (cardId, cardElement) => {
+    return api.deleteCard(cardId).then(() => {
+      cardElement.remove();
+    });
+  }
+);
+
+//USER INFO FUNCTIONS
+
+//get user info
 const userInfo = new UserInfo({
   name: ".profile__title",
   description: ".profile__description",
+  avatarSelector: ".profile__image",
 });
 
+api
+  .getUserInfo()
+  .then((userData) => {
+    userInfo.setUserInfo({
+      name: userData.name,
+      description: userData.about,
+      avatar: userData.avatar,
+    });
+  })
+  .catch((err) => console.error(err));
+
+//edit profile functionality
 const editProfilePopup = new PopupWithForm(
   "#profile-edit-modal",
   handleProfileEditSubmit
 );
 editProfilePopup.setEventListeners();
 
+function handleProfileEditSubmit(data) {
+  api
+    .editUserInfo({
+      name: data.title,
+      about: data.description,
+    })
+    .then((userData) => {
+      userInfo.setUserInfo({
+        name: userData.name,
+        description: userData.about,
+        avatar: userData.avatar,
+      });
+    })
+    .catch((err) => console.error(`Failed to update users info: ${err}`))
+    .finally(() => {
+      editProfilePopup.setLoading(false);
+    });
+
+  editProfilePopup.close();
```

Comment:
* You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well